appsmithorg/appsmith
# fix: [Git] Avoid 404 when checking out a branch

## Description

After checkout, we will now check if the resource the user was accessing is available in the incoming branch. Instead of calling the apis to check this, we will listen to the success action and then handle check if the current resource is still available in the branch. If not, we will navigate the user to the home page of the app so that they do not see a 404 error


> Don't show a 404 error when a resource is not available in the checked out branch, instead take them to the home page of the app

Fixes #17234
Fixes #20883


Media



## Type of change

- Bug fix (non-breaking change which fixes an issue)


## How Has This Been Tested?

- Manual
   - Have a git connected app
   - Create a new branch 
   - Create a new API/Query/Page on the new branch
   - Switch back to the original branch
   - Test: The app should not show 404 error but be navigated to the home page of the app

- Cypress
Updated the existing cypress tests that avoided the error to make sure they test the fix instead

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity


### QA activity:
